ui/react-app/src/utils/url-builder.ts

```diff
@@ -0,0 +1,17 @@
+const apiPrefix = '/api/v2';
+
+export type URLParams = {
+  resource: string;
+  queryParams?: URLSearchParams;
+  apiPrefix?: string;
+};
+
+export default function buildURL(params: URLParams): string {
+  let url = params.apiPrefix === undefined ? apiPrefix : params.apiPrefix;
+  url = `${url}/${params.resource}`;
+
+  if (params.queryParams !== undefined) {
+    url = `${url}?${params.queryParams.toString()}`;
+  }
+  return url;
+}
```

Comment:
I feel like we can simplify this a tad by using de-structuring with a default value assignment (assuming `apiPrefix` was renamed to `API_PREFIX` to denote a const).

```suggestion
export default function buildURL({ apiPrefix = API_PREFIX, resource, queryParams }: URLParams): string {
  let url = `${apiPrefix}/${resource}`;

  if (queryParams !== undefined) {
    url = `${url}?${queryParams.toString()}`;
  }

  return url;
}
```

Reply:
ah yeah indeed. Thanks for proposition !